File: src/main/java/com/example/storage/repository/IssueRepository.java
package com.example.storage.repository;

import com.example.storage.entity.Equipment;
import com.example.storage.entity.Issue;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface IssueRepository extends JpaRepository<Issue, Long> {
    Optional<Issue> findByEquipmentAndReturnedAtIsNull(Equipment equipment);
}

File: src/main/kotlin/ru/nto/storage/repository/IssueRepository.kt
package ru.nto.storage.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.nto.storage.entity.Equipment
import ru.nto.storage.entity.Issue

interface IssueRepository : JpaRepository<Issue, Long> {
    fun findByEquipmentAndReturnedAtIsNull(equipment: Equipment): Issue?
}
